File: binary_search.js
function binarySearch(array, target) {
    if (array.length === 0) return -1;

    const mid = Math.floor(array.length / 2);
    const left = array.slice(0, mid);
    const right = array.slice(mid + 1, array.length);

    if (array[mid] === target) {
        return mid;
    } else if (target < array[mid]) {
        return binarySearch(left, target);
    } else {
        let searchResult = binarySearch(right, target);
        return searchResult === -1 ? -1 : searchResult + mid + 1;
    }
}

File: binary_search.rb
def search(nums, target, left = 0, right = nums.length - 1)
    return -1 if left > right
    mid = (left + right) / 2
    case target <=> nums[mid]
    when 0
        return mid
    when 1
        return search(nums, target, mid + 1, right)
    when -1
        return search(nums, target, left, mid - 1)
    end
end